### ISSUE TITLE
Unique TEID per S8 session

### ISSUE THREAD

**Issue by adityawasudeo** [magma/magma#8313]
Most IPX expect a public IP address which necessitates a public IP assigned to every AGW. FFi proposes to build a service to dynamically map te_id -> sgw_ip eliminating this requirement. More details: https://github.com/magma/magma/discussions/8109

**Comment by ulaskozat** [comment]
This is actually at finer granularity at bearer level not even session. 

@uri200 we discussed unique TEID across AGWs  in the context of dedicated bearers, but IPX deployments that introduce a middlebox between SGW and PGW rely on unique TEIDs in a network. We need to unshelf some of the earlier design discussions and deliver this feature.

**Comment by ulaskozat** [comment]
@uri200 tentatively assigning 10 SWE days as estimate (3 days for scoping and 7 days for development and testing). Please adjust the estimate if you think otherwise.

**Comment by uri200** [comment]
Hi @ulaskozat  I was not part of this meeting so I am not sure what is the resolution for #8109. Does it mean that there will a new `on path device` on the data plane that will keep a map between teids and PGW?

Can we start working/defining this `on path device`. Using current implementation (Random Assignment) the system will work. So the creation of this `on path device` can be independent, and start in parallel of the new way to provide unique teids.



Also note that besides this new unique TEIDs service, we need to refactor what we already have implemented, which will increase the effort

Note there is now a design document (option 3 will be implemented)
https://docs.google.com/document/d/1TwHFv9nWW4cArZA5e6P6ik60NxAB75KK7SOFamEpD5s/edit?usp=sharing

**Comment by lpavao** [comment]
Requirement for both gigsky and dish. 
Without this they need 1 IPV4 address per AGW, will not be able to scale without this.
